vnext/Microsoft.ReactNative/IReactPropertyBag.idl

```diff
@@ -62,6 +62,9 @@ namespace Microsoft.ReactNative
       "It returns null if the property did not exist.\n"
       "If the new value is null, then the property is removed.")
     Object Set(IReactPropertyName name, Object value);
+
+    DOC_STRING("Gets the list of property names in the property bag")
+    IVector<IReactPropertyName> Names { get; };
```

Comment:
>Names [](http://example.com/codeflow?start=32&length=5)

I wonder why this change is needed? It would be better if we do not add it. Since the PropertyBag can be modified from multiple threads, I afraid that information from this property can be misused and cause multi-threaded bugs which cannot be found at development time, but hit by customers.
In any case, since taking list of names is not a trivial operation, it must not be a property, it must be a method.

Reply:
I found how it is used - to copy all properties from one PropertyBag to another. It is a valid operation that we may need in some other contexts. 
Would it better to have a new CopyFrom operation that adds all properties from one property bag to another atomically? I.e. using one scoped_lock for both mutexes in the source and target property bag.

Reply:
yeah it was either Names + manually copy, or a CopyFrom; I'll change it